app/Http/Controllers/CartController.php

```diff
@@ -54,29 +54,8 @@ public function addToCart(Request $request)
 
         Cart::add($slug, $qty, $attribute);
 
-
         $this->_setTaxAmount($slug, $qty);
```

Comment:
First of all sorry seems very complicated code.  

    $this->setTaxAmount($slug, $qty, $attribute);

And you want to replace `setTaxAmount`method as below:

     /**
     * Set the Tax Amount for the Product
     * 
     * @param string $slug
     * @param int $qty
     * @param array $attributes
     * @return self $this
     */
    private function _setTaxAmount($slug, $qty = 1, $attributes = [])
    {
        $productModel = $this->repository->findBySlug($slug);
        $isTaxEnabled = $this->configurationRepository->getValueByKey('tax_enabled');

        if ($isTaxEnabled && $productModel->is_taxable) {
            $percentage = $this->configurationRepository->getValueByKey('tax_percentage');
            
            if(null !== $attributes) {
                foreach($attributes as $attributeId => $productId) {
                    $productModel = $this->repository->find($productId);
                }
            }
            
            $taxAmount = ($percentage * $productModel->price / 100) * $qty;
            Cart::hasTax(true);
            Cart::updateProductTax($slug, $taxAmount);
        }

        return $this;
    }


Controller should look like below

    /**
     *
     * Add To Cart Product
     *
     * @param Request $request
     * @return \Illuminate\Http\RedirectResponse
     */
    public function addToCart(Request $request)
    {
        $slug = $request->get('slug');

        $qty = abs($request->get('qty', 1));
      
        $attribute = $request->get('attribute', null);

        if (!Cart::canAddToCart($slug, $qty, $attribute)) {
            return redirect()->back()
                        ->with(
                            'errorNotificationText',
                            'Not Enough Qty Available. Please with less qty or Contact site Administrator!'
                        );
        }

        Cart::add($slug, $qty, $attribute);
        $this->setTaxAmount($slug, $qty);

        return redirect()->back()->with('notificationText', 'Product Added to Cart Successfully!');
    }